### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by sekkit** [threestudio-project/threestudio#54]
Since im using UE5 in Windows, it's cool if threestudio could run under Windows.

**Comment by mischaschaub** [comment]
YES YES YES, this would be awfully great!

**Comment by thuliu-yt16** [comment]
Maybe you could first try WSL. The objective of threestudio now is to integrate most methods, so supporting new platform is of low priority.

**Comment by mischaschaub** [comment]
Exactly, this would be so helpful. There is a whole group of VoC users and many of their best people tried ot compile the WSL, but no chance. It would be ever so helpful if you could create such a file.

**Comment by claforte** [comment]
I think all core researchers are using Linux. If you want Windows support, I recommend you try to do it yourselves.

**Comment by shinya7y** [comment]
I have installed threestudio with WSL2.
I hope you find the following article helpful. Translation in Chrome generally works.
https://zenn.dev/shinya7y/articles/threestudio-prolificdreamer

**Comment by shengyu-meng** [comment]
> I have installed threestudio with WSL2. I hope you find the following article helpful. Translation in Chrome generally works. https://zenn.dev/shinya7y/articles/threestudio-prolificdreamer

I followed your guide and succeed! But sadly WSL2 only support OpenGL early than 4.1, so some main features, such as Fantasia3D / mesh exporter don't work😥.

**Comment by thuliu-yt16** [comment]
Hi @shengyu-meng, it is very nice! 

I guess for the mesh exporter, you can use the `cuda` context_type if opengl is uncompatible.

**Comment by shengyu-meng** [comment]
> Hi @shengyu-meng, it is very nice!
> 
> I guess for the mesh exporter, you can use the `cuda` context_type if opengl is uncompatible.

Thanks for your suggestion! So is it any chance to run Fantasia3D without openGL?  🥰

**Comment by IhnoL** [comment]
I used the docker method inside of WSL2 which seems to work. The exported meshes looked not so great, maybe due to the cuda exporter instead of opengl. I did not test Fantasia3D.

Docker and Nvidia Container Toolkit need to be installed: https://docs.nvidia.com/cuda/wsl-user-guide/index.html

**Comment by OmidGhotbi** [comment]
It is working with no issue on Windows, I managed to install it directly in Windows as well as using WSL you can use this tutorial for WSL installation https://www.youtube.com/watch?v=jaRr5W80N8E
the process is almost the same for Windows directly. i used cmd python 3.10 Pytorch 2.2.0 cuda 12.1
Visuald studio build tools and vc++ runtime should be installed then you need to install cuda toolkit from Nvidia website and used the same version of cuda 12.1
now you need to set venv like this "python -m venv .venv" then inside venv go to "cd /d C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build" and run "vcvars64.bat" then get back to threestudio folder and "set DISTUTILS_USE_SDK=1", install Pytorch 2.2.0 for cuda 121"pip install torch torchvision --index-url https://download.pytorch.org/whl/cu121" now install requirements pip install -r requirements.txt and that's all.
Good luck.
